src/gfn/utils/common.py

```diff
@@ -1,23 +1,172 @@
+import os
 import random
+import threading
+import warnings
+from typing import Callable, Tuple
 
 import numpy as np
 import torch
+import torch.distributed as dist
+
+# -----------------------------------------------------------------------------
```

Comment:
the changes in common.py are helpful in a distributed settings. Should they go in a distributed module (just a file in utils called `distributed.py`)?

Reply:
I'm not sure I totally understand this one - the logic here is `if distributed: do distributed things, else: do non-distributed things` -- it means the user can just `set_seed(number)` and it will work regardless if the code is being run in the distributed or non-distributed context.

Reply:
I see, sounds good then